src/Fixer/LanguageConstruct/ExplicitIndirectVariableFixer.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\LanguageConstruct;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\VersionSpecification;
+use PhpCsFixer\FixerDefinition\VersionSpecificCodeSample;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Filippo Tessarotto <zoeslam@gmail.com>
+ */
+final class ExplicitIndirectVariableFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Add curly braces to indirect variables to make them clear to understand. Requires PHP >= 7.0.',
+            [
+                new VersionSpecificCodeSample(
+<<<'EOT'
+<?php
+echo $$foo['bar'];
+echo $foo->$bar['baz'];
+
+EOT
+,
+                    new VersionSpecification(70000)
+                ),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return PHP_VERSION_ID >= 70000 && $tokens->isTokenKindFound(T_VARIABLE);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        foreach ($tokens as $index => $token) {
```

Comment:
not whole collection is traversed when you add tokens during process
I would suggest to iterate backwards

https://3v4l.org/sTQ0l

Reply:
I understand and will change this soon, but I'm unable to reproduce the issue and thus make a test.
I've tried:
```php
[
    '<?php' . str_repeat(' ${$foo};', 3000),
    '<?php' . str_repeat(' $$foo;', 3000),
],
```
And the test pass (after 30 seconds and thousands assertions).

Reply:
because of such issues, we become extra defensive in Tokens methods when handling with indexes, but imagine that later one will change implementation inside this loop (in fixer) and start doing sth fancy with index, then it could bring the real problem to the table (as we don't want to have such expensive test in codebase). Changing it now, when fixer is simple, is cheap